File: src/main/java/org/dementhium/content/areas/CoordinateEvent.java
package org.dementhium.content.areas;

import org.dementhium.model.Location;
import org.dementhium.model.Mob;

/**
 *
 * @author 'Mystic Flow <Steven@rune-server.org>
 *
 */
public abstract class CoordinateEvent {

	private Mob mob;
	
	private int x;
	private int y;
	private int lengthX;
	private int lengthY;

	public boolean areaObject = false;

	public CoordinateEvent(Mob mob, int x, int y) {
		this.mob = mob;
		this.x = x;
		this.y = y;
		areaObject = false;
	}
	
	public CoordinateEvent(Mob mob, int x, int y, int lengthX, int lengthY) {
		this.mob = mob;
		this.x = x;
		this.y = y;
		this.lengthX = lengthX;
		this.lengthY = lengthY;
		areaObject = true;
	}


	public boolean atArea() {
		Location loc = mob.getLocation();
		return loc.getX() == x && loc.getY() == y;
	}

	public boolean inArea() {
		Location loc = mob.getLocation();
		return loc.getX() >= x - lengthX && loc.getY() >= y - lengthY && loc.getX() <= x + lengthX && loc.getY() <= y + lengthY;
	}

	public abstract void execute();
}

File: src/main/java/org/dementhium/content/areas/CoordinateEvent.kt
package org.dementhium.content.areas

import org.dementhium.model.Mob

abstract class CoordinateEvent @JvmOverloads constructor (
    val mob: Mob,
    val x: Int,
    val y: Int,
    val lengthX: Int = 0,
    val lengthY: Int = 0
) {

    var areaObject: Boolean = !(lengthX == 0 || lengthY == 0)

    fun atArea(): Boolean {
        val loc = mob.location
        return loc.x == x && loc.y == y
    }

    fun inArea(): Boolean {
        val loc = mob.location
        return loc.x >= x - lengthX && loc.y >= y - lengthY && loc.x <= x + lengthX && loc.y <= y + lengthY
    }

    abstract fun execute()
}